app/services/providers/copy_to_recruitment_cycle_service.rb

```diff
@@ -8,12 +8,12 @@ def initialize(copy_course_to_provider_service:, copy_site_to_provider_service:,
       @logger = logger || Logger.new("/dev/null")
     end
 
-    def execute(provider:, new_recruitment_cycle:, force: false)
+    def execute(provider:, new_recruitment_cycle:, force:)
```

Comment:
I think the intention of `force` was just to allow us to roll providers that haven't been rolled over because they weren't rollable. I don't think we should overload this, especially with it being a boolean or a symbol. I'm not really sure what we're trying to achieve with the :provider and :provider_and_courses. It seems like we're trying to make this do too much.

Reply:
"1DX" has 86 courses from last cycle but have not ran any of them previously

`:provider` will be the 
> I think the intention of force was just to allow us to roll providers that haven't been rolled over because they weren't rollable.

`:provider_and_courses.` will be the above as well as actually rolling over the course that has not be ran previously


otherwise, on without any code change

```ruby
old_provider = RecruitmentCycle.current.previous.providers.find_by(code: "1DQ")

    copy_provider_to_recruitment_cycle = Providers::CopyToRecruitmentCycleService.new(
      copy_course_to_provider_service: nil,
      copy_site_to_provider_service: Sites::CopyToProviderService.new,
      logger: Logger.new(STDOUT),
    )

    copy_provider_to_recruitment_cycle.execute(
      provider: old_provider, new_recruitment_cycle: RecruitmentCycle.current, force: true,
    )

    new_provider = RecruitmentCycle.current.providers.find_by(code: "1DQ")

    old_provider.courses.each do |course|
      Courses::CopyToProviderService.new(
        sites_copy_to_course: Sites::CopyToCourseService.new,
        enrichments_copy_to_course: Enrichments::CopyToCourseService.new,
      ).execute(course: course, new_provider: new_provider, force: true)    
    end 

```

To be ran arbitrary per supports needs,

Reply:
I still think this is overloading 'force'. It's taking a lot of explanation which means the intention isn't clear. It's probably a mistake rolling the courses over at all in the provider service, that's where the complexity of all of this is coming from.